src/domainobjects/instrument.py

```diff
@@ -3,32 +3,59 @@
 
 class Instrument(Generatable):
     
-    def generate(self, record_count, custom_args):        
-        records = []
+    def generate(self, record_count, custom_args):
+        config = self.get_object_config()
+        records_per_file = config['max_objects_per_file']
+        file_num = 1
+
+        database = self.get_database()
         
+        records = []
+        current_tickers = {}
+
         for i in range(0, record_count):            
             asset_class = self.generate_asset_class()         
-            ticker = self.generate_currency() if asset_class == 'Cash' else self.generate_ticker()            
-            exchange_country = '' if asset_class == 'Cash' else self.generate_exchange_country()
-            exchange_code = '' if asset_class == 'Cash' else exchange_country[0]
-            coi = '' if asset_class == 'Cash' else exchange_country[1]            
-            cusip = '' if asset_class == 'Cash' else str(self.generate_random_integer(length=9))
-            isin = '' if asset_class == 'Cash' else self.generate_isin(coi, cusip)
-            ric = '' if asset_class == 'Cash' else self.generate_ric(ticker, exchange_code)
-            sedol = '' if asset_class == 'Cash' else self.generate_random_integer(length=7)
-                
+            ticker = self.generate_ticker()
+            coi = self.generate_coi()
+            exchange_code = self.\
+                generate_sequential_exchange_code(current_tickers, ticker)
+            cusip = str(self.generate_random_integer(length=9))
+            isin = self.generate_isin(coi, cusip)
+            ric = self.generate_ric(ticker, exchange_code)
+            sedol = self.generate_random_integer(length=7)
+            j = i+1
```

Comment:
Why add this variable? I ain't the biggest fan of singletons. What does i and j stand for as well? I assume i is a counter, so why we need to store it as i+1. Naming the variables could help here